### PR TITLE
Bug/jenkins 45754 GitHub enterprise url validation 2

### PR DESCRIPTION
**Pull request by cliffmeyers**
# Description
See [JENKINS-45754](https://issues.jenkins-ci.org/browse/JENKINS-45754).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by cliffmeyers** [comment]
Completely bogus URL triggers more generic / detailed error messaging:
validate-ghe-url-1-very-bogus-url

URL pointing to real server but not GitHub server triggers more specific error message:
validate-ghe-url-2-server-not-github

URL pointing to a valid GitHub server but not API endpoint triggers more specific error message:
validate-ghe-url-3-server-github-bad-endpoint

**Comment by i386** [comment]
Nice tests 🐝

**Comment by i386** [comment]
@vivek mind giving a review?

File: blueocean-dashboard/src/main/js/creation/github-enterprise/api/CreateServerError.js [hunk 1]
```diff
@@ -6,7 +6,8 @@ class CreateServerError extends ServerError {
     _handleCustomErrors(errors) {
         this.duplicateName = errors.some(err => err.field === 'name' && err.code === 'ALREADY_EXISTS');
         this.duplicateUrl = errors.some(err => err.field === 'apiUrl' && err.code === 'ALREADY_EXISTS');
-        this.invalidUrl = errors.some(err => err.field === 'apiUrl' && err.code === 'INVALID');
+        this.invalidServer = errors.some(err => err.field === 'apiUrl' && err.code === 'INVALID' && err.message.indexOf('check hostname') !== -1);
+        this.invalidApiUrl = errors.some(err => err.field === 'apiUrl' && err.code === 'INVALID' && err.message.indexOf('check path') !== -1);
```
Comment (vivek) [inline blueocean-dashboard/src/main/js/creation/github-enterprise/api/CreateServerError.js]: We are depending on error message text to flag error, to protect against backend message change, better to have tests for it, ideally in ath.

File: acceptance-tests/src/test/java/io/blueocean/ath/live/GithubEnterpriseCreationTest.java [hunk 2]
```diff
@@ -68,10 +68,10 @@ public void testGitHubEnterpriseCreation_addNewGitHubServer() throws IOException
         dialog.findFormErrorMessage("enter a valid URL");
         // server-side URL validation
         dialog.enterServerName(serverName);
-        dialog.enterServerUrl("foo");
+        dialog.enterServerUrl("http://www.google.com");
         dialog.waitForErrorMessagesGone();
         dialog.clickSaveServerButton();
-        dialog.findFormErrorMessage("Could not connect");
+        dialog.findFormErrorMessage("Invalid server");
```
Comment (vivek) [inline acceptance-tests/src/test/java/io/blueocean/ath/live/GithubEnterpriseCreationTest.java]: test explicitly for `Check hostname` and `Check path`?

File: blueocean-dashboard/src/main/js/creation/github-enterprise/api/CreateServerError.js (hunk 1 above)
Comment (cliffmeyers) [inline blueocean-dashboard/src/main/js/creation/github-enterprise/api/CreateServerError.js]: Fixed in 444db44

File: acceptance-tests/src/test/java/io/blueocean/ath/live/GithubEnterpriseCreationTest.java (hunk 2 above)
Comment (cliffmeyers) [inline acceptance-tests/src/test/java/io/blueocean/ath/live/GithubEnterpriseCreationTest.java]: Fixed in 444db44

**Comment by cliffmeyers** [comment]
@vivek those scenarios are now covered by ATH. It requires pointing to github.beescloud.com which I was hoping to avoid but hopefully the server remains stable.

**Comment by michaelneale** [comment]
LGTM - I think viveks comments addressed - and build happy

**Comment by vivek** [comment]
LGTM 🐝

**Pull request merged**